target/linux/ramips/base-files/lib/ramips.sh

```diff
@@ -85,8 +85,11 @@ ramips_board_detect() {
 	*"C108")
 		name="c108"
 		;;
-	*"C20")
-		name="c20"
+	*"C20 v1")
+		name="c20v1"
+		;;
+	*"C20 v4")
```

Comment:
Drop the C20 v4 here and use the compatible string as boardname in all files (tplink,c20v4).